src/github.com/mozilla-services/pushgo/simplepush/app.go

```diff
@@ -294,7 +293,11 @@ func (a *Application) closeClients() {
 	}
 }
 
-func (a *Application) CloseOnce() error {
+func (a *Application) Close() error {
+	return a.closeOnce.Do(a.close)
+}
+
+func (a *Application) close() error {
```

Comment:
Agreed; there's a lot of indirection. `if !a.closeOnce.IsDone() { ... a.closeOnce.Close() }` doesn't look safe for concurrent calls, but I think a closure would work.